FLT/HIMExperiments/module_topology.lean

```diff
@@ -145,29 +146,37 @@ example (ι : Type*) [Finite ι] :
+  rw [← continuous_iff_le_induced, show i = LinearMap.lsmul A A (i 1) by ext; simp]
   exact continuous_const.mul continuous_id
 
-lemma LinearMap.continuous_on_prod (f : (M × N) →ₗ[A] A) :  @Continuous _ _ (@instTopologicalSpaceProd M N (Module.topology A) (Module.topology A)) _ f := by
-  have : ⇑f = fun ⟨m, n⟩ ↦ f (⟨m, 0⟩) + f (⟨0, n⟩) := by
-    ext x
-    simp only
-    have : x = (x.1, 0) + (0, x.2) := by simp
-    nth_rewrite 1 [this]
-    apply LinearMap.map_add
-  rw [this]
-  simp only
-  apply @Continuous.add _ _ (@instTopologicalSpaceProd M N (Module.topology A) (Module.topology A)) _ _ _ (fun x ↦ (fun m ↦ f (m, 0)) x.1) (fun x ↦ (fun n ↦ f (0, n)) x.2)
-  . apply @Continuous.fst' _ _ _ (Module.topology A) (Module.topology A) _ (fun m ↦ f (m, 0))
-    nth_rewrite 2 [Module.topology_self A]
-    exact Module.continuous_linear A ({toFun := fun m ↦ f (m, 0), map_add' := by
-    {intro x y; rw [← LinearMap.map_add, Prod.mk_add_mk, zero_add]}, map_smul' := by
-    {intro m x; rw [← LinearMap.map_smul, RingHom.id_apply, Prod.smul_mk, smul_zero]}})
-  . apply @Continuous.snd' _ _ _ (Module.topology A) (Module.topology A) _ (fun n ↦ f (0, n))
-    nth_rewrite 2 [Module.topology_self A]
-    exact Module.continuous_linear A ({toFun := fun n ↦ f (0, n), map_add' := by
-    {intro x y; rw [← LinearMap.map_add, Prod.mk_add_mk, add_zero]}, map_smul' := by
-    {intro m x; rw [← LinearMap.map_smul, RingHom.id_apply, Prod.smul_mk, smul_zero]}})
+lemma Module.continuous_linear_from_ring (e : A →ₗ[A] M) :
+    @Continuous A M _ (Module.topology A) e := by
+  nth_rw 1 [Module.topology_self A]
+  exact Module.continuous_linear A e
+
+lemma Module.continuous_linear_to_ring (e : M →ₗ[A] A) :
+    @Continuous M A (Module.topology A) _ e := by
+  nth_rw 2 [Module.topology_self A]
+  exact Module.continuous_linear A e
+
+lemma LinearMap.continuous_on_prod (f : (M × N) →ₗ[A] A) :
+    @Continuous _ _ (@instTopologicalSpaceProd M N (Module.topology A) (Module.topology A)) _ f := by
+  let _τM : TopologicalSpace M := Module.topology A
+  let _τN : TopologicalSpace N := Module.topology A
+  suffices Continuous fun (⟨m, n⟩ : M × N) ↦ f (⟨m, 0⟩) + f (⟨0, n⟩) by
+    convert this
+    simp_rw [← LinearMap.map_add, Prod.mk_add_mk, add_zero, zero_add]
+  apply Continuous.add
+  . apply @Continuous.fst' _ _ _ _ _ _ (fun m ↦ f (m, 0))
+    exact Module.continuous_linear_to_ring A
+      ({toFun := fun m ↦ f (m, 0),
+        map_add' := by {intro x y; rw [← LinearMap.map_add, Prod.mk_add_mk, zero_add]},
+        map_smul' := by {intro m x; rw [← LinearMap.map_smul, RingHom.id_apply, Prod.smul_mk, smul_zero]}})
+  . apply @Continuous.snd' _ _ _ _ _ _ (fun n ↦ f (0, n))
+    exact Module.continuous_linear_to_ring A
+      ({toFun := fun n ↦ f (0, n),
+        map_add' := by {intro x y; rw [← LinearMap.map_add, Prod.mk_add_mk, add_zero]},
+        map_smul' := by {intro m x; rw [← LinearMap.map_smul, RingHom.id_apply, Prod.smul_mk, smul_zero]}})
```

Comment:
```suggestion
        map_smul' := by 
          intro m x; rw [← LinearMap.map_smul, RingHom.id_apply, Prod.smul_mk, smul_zero]})
```
Here and below can you stick to <= 100 chars in a line? (the grey line in VS Code shows you where to stop)